Packs/ApiModules/Scripts/AggregatedCommandApiModule/AggregatedCommandApiModule.py

```diff
@@ -0,0 +1,1219 @@
+        """
+        if not self.requested:
+            demisto.debug("No specific brands provided; will run on all available brands.")
+            return []
+
+        brands_to_execute = list(set(self.requested) & set(self.enabled))
+        if not brands_to_execute:
+            raise DemistoException(
+                "None of the provided brands correspond to an enabled integration instance. "
+                "Please ensure brand names are correct and instances are enabled."
+            )
+        demisto.debug(f"[BrandManager] to_run={brands_to_execute}")
+        return brands_to_execute
+
+    @cached_property
+    def missing(self) -> list[str]:
+        """
+        Returns a list of missing brands from the given brands.
+        If no brands are given, returns empty list.
+        Caches the result to avoid redundant calculations.
+        """
+        if not self.requested:
+            return []
+
+        missing_brands = list(set(self.requested) - self.enabled)
+        demisto.debug(f"[BrandManager] missing={missing_brands}")
+        return missing_brands
+
+    def unsupported_external(self, commands: list[list["Command"]]) -> list[str]:
+        """
+        Returns a list of unsupported enrichment brands to run on from the given brands.
+        If no brands are given, returns empty list.
+        Caches the result to avoid redundant calculations.
+        """
+        if not self.requested:
+            demisto.debug("No specific brands provided; will run on all available brands.")
+            return []
+        non_external_brands = {
+            command.brand
+            for command_list in commands
+            for command in command_list
+            if command.command_type != CommandType.EXTERNAL and command.command_type != CommandType.BUILTIN
+        }
+        demisto.debug(f"[BrandManager] non_external_brands={non_external_brands}")
+        external_brands = set(self.requested) - non_external_brands
+        demisto.debug(f"[BrandManager] external_brands={external_brands}")
+        return list(external_brands - set(self.enabled))
+
+
+# --- Main Framework Abstraction ---
+class AggregatedCommand(ABC):
+    def __init__(
+        self,
+        args: dict,
+        brands: list[str],
+        verbose: bool,
+        commands: list[list[Command]] | None = None,
+    ):
+        """
+        Initializes the module.
+
+        Args:
+            args (dict[str, Any]): The arguments from `demisto.args()`.
+            brands (list[str]): A list of specific integration brands to run.
+            verbose (bool): If True, include detailed command outputs.
+            commands (list[list[Command]]): List of batches of commands to run.
+        """
+        self.args = args
+        self.brands = brands
+        self.verbose = verbose
+        self.commands = commands or []
+        self.brand_manager = BrandManager(brands)
+
+    @cached_property
+    def unsupported_enrichment_brands(self) -> list[str]:
```

Comment:
```suggestion
    def unsupported_brands(self) -> list[str]:
```
is this enrichment specific? We are in the base class here

Reply:
I will move it to the Reputation it specific for unsupported enrichment brands and not internal brands